Sources/NIO/SocketChannel.swift

```diff
@@ -520,6 +548,47 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
         }
         return readResult
     }
+    
+    private struct ControlMessageReceiver {
```

Comment:
If we add a collection type for control messages, this becomes a simple struct that takes that collection as input and consumes it, turning it into simple values.

Reply:
For discussion - I've kept the collection in the recvmsg function for now to minimise the chances of the control bytes pointer escaping this scope.

Reply:
I think there's no risk: so long as we have this essentially "consume" the collection in its init we're safe, there's only one place to audit.

Reply:
I'd also like to see this move out into another file with the collection we just created.